### DISCUSSION TITLE
Single Grafana Agent installation vs multiple

### DISCUSSION THREAD

**Discussion by ashikhman** [grafana-cold-storage/agent#3644] [Q&A]
We are using Grafana Agent Operator to collect logs and metrics, and custom agent deployment for traces. The operator creates these resources:
1. Agent stateful set for metrics scraping
2. Agent daemonset for node exporter
3. Agent deployment for event handler
4. Agent daemonset for logs
5. Agent deployment for traces (installed through custom Helm chart)

Wouldn't it better to create single daemonset to scrape/receive logs, metrics and traces?

**Comment by nmabhinandan** [comment]
this and is the grafana-agent helm chart which deploys DaemonSet by default and uses flow mode by default sufficient for all three use cases?

**Comment by Charlotte-br560** [comment]
Agreed, consolidating into a single daemonset for unified scraping/receiving of logs, metrics, and traces could streamline management and resource utilization. This approach simplifies deployment and maintenance overhead.
